### PR TITLE
docs: add example for: awscc_rds_db_parameter_group

### PR DESCRIPTION
**Pull request by ishwar-amzn**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request
* The resources and data sources in this provider are generated from the CloudFormation schema, so they can only support the actions that the underlying schema supports. For this reason submitted bugs should be limited to defects in the generation and runtime code of the provider. Customizing behavior of the resource, or noting a gap in behavior are not valid bugs and should be submitted as enhancements to AWS via the CloudFormation Open Coverage Roadmap.

Relates OR Closes #0000

### CODE REVIEW

File: examples/resources/awscc_rds_db_parameter_group/awscc_rds_db_parameter_group_create_before_destroy.tf [hunk 1]
```diff
@@ -0,0 +1,24 @@
+resource "awscc_rds_db_parameter_group" "this1" {
```
Comment (wellsiau-aws) [inline examples/resources/awscc_rds_db_parameter_group/awscc_rds_db_parameter_group_create_before_destroy.tf:1]:
```suggestion
resource "awscc_rds_db_parameter_group" "this" {
```

File: examples/resources/awscc_rds_db_parameter_group/awscc_rds_db_parameter_group_create_before_destroy.tf [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    "character_set_server" = "utf8"
+    "character_set_client" = "utf8"
+  }
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "awscc_rds_db_instance" "this" {
```
Comment (wellsiau-aws) [inline examples/resources/awscc_rds_db_parameter_group/awscc_rds_db_parameter_group_create_before_destroy.tf:15]: this resource is not required to complete the example, so I suggest to skip / remove this resource block

File: templates/resources/awscc_rds_db_parameter_group.md.tmpl [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+{{ .Description | trimspace }}
+
+## Example Usage
+
+### Basic example
+To create a simple db subnet group
+{{ tffile (printf "examples/resources/%s/awscc_rds_db_parameter_group.tf" .Name)}}
+
+### To create a db subnet group with tags
+To enable Storage Autoscaling with instances that support the feature, define the max_allocated_storage argument higher than the allocated_storage argument. Terraform will automatically hide differences with the allocated_storage argument value if autoscaling occurs.
```
Comment (wellsiau-aws) [inline templates/resources/awscc_rds_db_parameter_group.md.tmpl:19]: this description is not relevant to the example provided

File: templates/resources/awscc_rds_db_parameter_group.md.tmpl [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+{{ tffile (printf "examples/resources/%s/awscc_rds_db_parameter_group_tags.tf" .Name)}}
+
+### To create a db subnet group with `create_before_destroy` Lifecycle Configuration example
+You can specify the manage_master_user_password attribute to enable managing the master password with Secrets Manager. You can also update an existing cluster to use Secrets Manager by specify the manage_master_user_password attribute and removing the password attribute (removal is required).
```
Comment (wellsiau-aws) [inline templates/resources/awscc_rds_db_parameter_group.md.tmpl:23]: this description is not relevant to the example provided

**Review by wellsiau-aws — CHANGES_REQUESTED** [review]

File: examples/resources/awscc_rds_db_parameter_group/awscc_rds_db_parameter_group_create_before_destroy.tf (hunk 1 above)
Comment (ishwar-amzn) [reply examples/resources/awscc_rds_db_parameter_group/awscc_rds_db_parameter_group_create_before_destroy.tf:1]: Corrected said changes

File: examples/resources/awscc_rds_db_parameter_group/awscc_rds_db_parameter_group_create_before_destroy.tf (hunk 2 above)
Comment (ishwar-amzn) [reply examples/resources/awscc_rds_db_parameter_group/awscc_rds_db_parameter_group_create_before_destroy.tf]: Removed suggested resource block

File: templates/resources/awscc_rds_db_parameter_group.md.tmpl (hunk 3 above)
Comment (ishwar-amzn) [reply templates/resources/awscc_rds_db_parameter_group.md.tmpl]: Corrected relevant description

File: templates/resources/awscc_rds_db_parameter_group.md.tmpl (hunk 4 above)
Comment (ishwar-amzn) [reply templates/resources/awscc_rds_db_parameter_group.md.tmpl]: Corrected relevant description

**Comment by wellsiau-aws** [comment]
@ishwar-amzn , can you rerun `make docs` to consolidate the changes, the check is failing because the final doc did not match the template file

**Review by wellsiau-aws — CHANGES_REQUESTED** [review]
please re-run `make docs`

**Comment by ishwar-amzn** [comment]
Sure @wellsiau-aws , I am running make docs but still getting error. Let me dive deep and solve all issues.

File: templates/resources/awscc_rds_db_parameter_group.md.tmpl
Comment (wellsiau-aws) [inline templates/resources/awscc_rds_db_parameter_group.md.tmpl:1]: this template file name should be just `rds_db_parameter_group.md.tmpl`

**Review by wellsiau-aws — CHANGES_REQUESTED** [review]

**Comment by marcosentino** [comment]
Thank you @ishwar-amzn for the contribution!

**Pull request merged**